pubsub/api/Pubsub.Samples.Tests/PullMessagesAsyncWithDeliveryAttemptsTest.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Xunit;
+
+[Collection(nameof(PubsubFixture))]
+public class PullMessagesAsyncWithDeliveryAttemptsTest
+{
+    private readonly PubsubFixture _pubsubFixture;
+    private readonly PublishMessagesAsyncSample _publishMessagesAsyncSample;
+    private readonly PullMessagesAsyncWithDeliveryAttemptsSample _pullMessagesAsyncWithDeliveryAttemptsSample;
+    private readonly CreateSubscriptionWithDeadLetterPolicySample _createSubscriptionWithDeadLetterPolicySample;
+
+    public PullMessagesAsyncWithDeliveryAttemptsTest(PubsubFixture pubsubFixture)
+    {
+        _pubsubFixture = pubsubFixture;
+        _publishMessagesAsyncSample = new PublishMessagesAsyncSample();
+        _pullMessagesAsyncWithDeliveryAttemptsSample = new PullMessagesAsyncWithDeliveryAttemptsSample();
+        _createSubscriptionWithDeadLetterPolicySample = new CreateSubscriptionWithDeadLetterPolicySample();
+    }
+
+    [Fact]
+    public async void PullMessagesAsyncWithDeliveryAttempts()
+    {
+        string randomName = _pubsubFixture.RandomName();
+        string topicId = $"testTopicForDeadLetterPolicyMessageSyncAck{randomName}";
+        string deadLetterTopicId = $"testTopicForDeadLetter{randomName}";
+        string subscriptionId = $"testSubscriptionDeadLetterPolicyForMessageSyncAck{randomName}";
+        var message = _pubsubFixture.RandomName();
+
+        _pubsubFixture.CreateTopic(topicId);
+        _pubsubFixture.CreateTopic(deadLetterTopicId);
```

Comment:
Do we really need a deadLetterTopic here? Our sample is just showing that there's a field called DeliveryAttempts. I believe a subscriptions without a dead letter topic is fine here.

Reply:
Yes we need `deadLetterTopic`  I just created Once in setup and re-used in other test as well.